src/Rings/msolve/msolve.jl

```diff
@@ -0,0 +1,213 @@
+import msolve_jll: libmsolve
+import Libdl: dlopen, dlsym, dlclose
+
+export msolve
+
+@doc Markdown.doc"""
+    function get_rational_parametrization(nr::Int32, lens::Array{Int32,1}, cfs::Ptr{BigInt})
+
+Construct the rational parametrization of the solution set computed via msolve.
+
+**Note**: This is an internal function and should only be used inside `msolve()`.
+"""
+function get_rational_parametrization(
+        nr::Int32,
+        lens::Array{Int32,1},
+        cfs::Ptr{BigInt}
+    )
+    C, x  = PolynomialRing(QQ,"x")
+    ctr   = 0
+
+    elim  = 0*x
+    for i in 1:lens[1]
+        elim  +=  BigInt(unsafe_load(cfs, i))*x^(i-1)
+    end
+    ctr += lens[1]
+
+    denom = 0*x
+    for i in 1:lens[2]
+        denom +=  BigInt(unsafe_load(cfs, i+ctr))*x^(i-1)
+    end
+    ctr +=  lens[2]
+
+    size  = nr-2
+    p = Array{PolyElem,1}(undef, size)
+    c = Array{BigInt,1}(undef, size)
+    k = 1
+    for i in 3:nr
+        p[k]  = 0*x
+        for j in 1:lens[i]-1
+            p[k]  +=  BigInt(unsafe_load(cfs, j+ctr))*x^(j-1)
+        end
+        c[k]  =   (-1) * BigInt(unsafe_load(cfs, lens[i]+ctr))
+        ctr   +=  lens[i]
+        k     +=  1
+    end
+
+    return elim, denom, p, c
+end
+
+@doc Markdown.doc"""
+    msolve(I, <keyword arguments>)
+
+Compute the solution set of the given ideal `I` if the ideal is zero dimensional.
+See [BES21](@cite) for more information.
+
```

Comment:
should at least the output be described? Complex roots with a precision of ... or so? Which complex field?

Reply:
Have a look, I tried to make the description more expressive.